prodsys/simulation/router.py

```diff
@@ -54,38 +54,43 @@ def route_request(self, processing_request: request.Request) -> Generator:
         """
         possible_resources = self.get_possible_resources(processing_request)
+        possible_resources_reached = []
         if not possible_resources:
             raise ValueError(f"No possible production resources found for request of product {processing_request.product.product_data.ID} and process {processing_request.process.process_data.ID}.")
         if not isinstance(processing_request, request.TransportResquest):
-            possible_resources = [
-                resource
-                for resource in possible_resources
-                if self.can_reach_resource(
-                    product=processing_request.product,
-                    resource=resource,
-                )
-            ]
-        if not possible_resources:
+            for resource in possible_resources:
+                if self.can_reach_resource(product=processing_request.product, resource=resource): 
+                    possible_resources_reached.append(resource)
+                else:
+                    break
+        else:
+            possible_resources_reached = possible_resources
+         
+        if not possible_resources_reached:
             raise ValueError(f"No possible transport resources found for request of product {processing_request.product.product_data.ID} and process {processing_request.process.process_data.ID} to reach any destinations from resource {processing_request.product.current_location.data.ID}.")
 
         while True:
-            free_resources = self.get_free_resources(possible_resources)
+            free_resources = self.get_free_resources(possible_resources_reached)
             self.routing_heuristic(free_resources)
-            # make timeout of 0 to make sure not two waiting requests are triggered at the same time and request the same resource
             yield processing_request.product.env.timeout(0)
             if free_resources:
                 break
             logger.debug({"ID": processing_request.product.product_data.ID, "sim_time": processing_request.product.env.now, "event": f"Waiting for free resources."})
             yield events.AnyOf(
                 processing_request.product.env,
-                [resource.got_free for resource in possible_resources],
+                [resource.got_free for resource in possible_resources_reached],
             )
             logger.debug({"ID": processing_request.product.product_data.ID, "sim_time": processing_request.product.env.now, "event": f"Free resources available."})
 
 
         if not free_resources:
             raise ValueError("No free resources available, Error in Event handling of routing to resources.")
         routed_resource = free_resources[0]
+        if isinstance(routed_resource, resources.TransportResource) and (isinstance(processing_request.process, process.RequiredCapabilityProcess) or isinstance(processing_request.process, process.LinkTransportProcess)):
```

Comment:
This is not very understandable. Please rename variables i and x and don't use a "path" dict feature which is intransparent...

Reply:
Commit 30